server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java

```diff
@@ -161,6 +161,9 @@ boolean useDelegate(String name) {
                 // well if using mmap.
                 case "tip":
                 case "dim":
```

Comment:
add a comment that this one can be removed when we are on Lucene 10 since it only applies to Lucene 8.x indices?